oneflow/user/kernels/dim_gather_kernel_util.h

```diff
@@ -43,17 +43,24 @@ template<DeviceType device_type, typename IN_T, typename IDX_T>
 struct DimGatherFunctor final {
   void operator()(ep::Stream* stream, const DimOpIndexNdHelper<IDX_T>& input_nd_helper,
                   const DimOpIndexNdHelper<IDX_T>& index_nd_helper, int ndim, int64_t elem_cnt,
-                  int32_t dim, const IDX_T* index, const IN_T* input, IN_T* output);
+                  int64_t dim_length, int32_t dim, const IDX_T* index, const IN_T* input,
+                  IN_T* output);
 };
 
 template<typename IN_T, typename IDX_T>
 OF_DEVICE_FUNC void DoDimGather(const DimOpIndexNdHelper<IDX_T>& input_nd_helper,
                                 const DimOpIndexNdHelper<IDX_T>& index_nd_helper, int ndim,
-                                int64_t elem_cnt, int32_t dim, const IDX_T* index,
-                                const IN_T* input, IN_T* output) {
+                                int64_t elem_cnt, int64_t dim_length, int32_t dim,
+                                const IDX_T* index, const IN_T* input, IN_T* output) {
   XPU_1D_KERNEL_LOOP(index_offset, elem_cnt) {
     IDX_T coordinate[kDimGatherMaxDimCount] = {0};
     const IDX_T x = index[index_offset];
+#ifdef __CUDA_ARCH__
+    assert(x < dim_length && "gather index is out of bounds");
```

Comment:
```suggestion
    assert(x < dim_length); // gather index is out of bounds
```

是不是用注释就可以了，这个字符串的作业也只是注释作用？

Reply:
好像不太一样，如果触发了这个检查会直接在命令行上抛出这个错误。

Reply:
不会吧，assert应该是没有打印的能力的。

这里的字符串只是一个表达式，表达式的值是一个有效的指针地址(总不为0)。所以assert到底值是多少，完全取决于dim_length，所以这个字符串我觉得就只是注释用了。

另外我搜了下代码里，.cu 文件里的assert都没这样加字符串的